* grobian@gentoo.org: allow for compilation on Solaris, lazy approach to
                      resolve a conflict with FS from sys/regset.h, fix
                      type u_int32_t into uint32_t.

type fixes are already in upstream git, the FS problem perhaps is also
resolved

--- include/cdio++/iso9660.hpp
+++ include/cdio++/iso9660.hpp
@@ -35,6 +35,13 @@
 #include <cstdlib>
 using namespace std;
 
+/* On Solaris, FS is defined to an integer which makes compilation fail
+ * since we use that as identifier below.  It gets included via the
+ * chain sys/signal.h -> sys/ucontext.h -> sys/regset.h. */
+#ifdef FS
+# undef FS
+#endif
+
 /** ISO 9660 class.
 */
 class ISO9660
--- src/cddb.h
+++ src/cddb.h
@@ -38,7 +38,7 @@
       the total length of the disk, and 
       the number of tracks.
 */
-u_int32_t cddb_discid(CdIo_t *p_cdio, track_t i_tracks);
+uint32_t cddb_discid(CdIo_t *p_cdio, track_t i_tracks);
 
 #ifdef HAVE_CDDB
 #include <cddb/cddb.h>
--- src/cddb.c
+++ src/cddb.c
@@ -48,7 +48,7 @@
       the total length of the disk, and 
       the number of tracks.
 */
-u_int32_t
+uint32_t
 cddb_discid(CdIo_t *p_cdio, track_t i_tracks)
 {
   int i,t,n=0;
